clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusteredTierActiveEntity.java

```diff
@@ -0,0 +1,683 @@
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.unmodifiableSet;
+import static org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.allInvalidationDone;
+import static org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.clientInvalidateAll;
+import static org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.clientInvalidateHash;
+import static org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.hashInvalidationDone;
+import static org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.serverInvalidateHash;
+import static org.ehcache.clustered.common.internal.messages.EhcacheMessageType.isLifecycleMessage;
+import static org.ehcache.clustered.common.internal.messages.EhcacheMessageType.isPassiveReplicationMessage;
+import static org.ehcache.clustered.common.internal.messages.EhcacheMessageType.isStateRepoOperationMessage;
+import static org.ehcache.clustered.common.internal.messages.EhcacheMessageType.isStoreOperationMessage;
+import static org.ehcache.clustered.server.ConcurrencyStrategies.DefaultConcurrencyStrategy.DATA_CONCURRENCY_KEY_OFFSET;
+import static org.ehcache.clustered.server.ConcurrencyStrategies.DefaultConcurrencyStrategy.DEFAULT_KEY;
+
+/**
+ * ClusteredTierActiveEntity
+ */
+public class ClusteredTierActiveEntity implements ActiveServerEntity<EhcacheEntityMessage, EhcacheEntityResponse> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ClusteredTierActiveEntity.class);
+  static final String SYNC_DATA_SIZE_PROP = "ehcache.sync.data.size.threshold";
+  private static final long DEFAULT_SYNC_DATA_SIZE_THRESHOLD = 4 * 1024 * 1024;
+
+  private final String storeIdentifier;
+  private final ServerStoreConfiguration configuration;
+  private final EhcacheEntityResponseFactory responseFactory;
+  private final ClientCommunicator clientCommunicator;
+  private final EhcacheStateService stateService;
+  private final IEntityMessenger entityMessenger;
+  private final ServerStoreCompatibility storeCompatibility = new ServerStoreCompatibility();
+  private final ConcurrentMap<ClientDescriptor, Boolean> connectedClients = new ConcurrentHashMap<>();
+  private final AtomicBoolean reconnectComplete = new AtomicBoolean(true);
+  private final AtomicInteger invalidationIdGenerator = new AtomicInteger();
+  private final ConcurrentMap<Integer, InvalidationHolder> clientsWaitingForInvalidation = new ConcurrentHashMap<>();
+  private final Set<UUID> trackedClients = Collections.newSetFromMap(new ConcurrentHashMap<>());
+  private final InvalidStoreException creationException;
+  private final ReconnectMessageCodec reconnectMessageCodec = new ReconnectMessageCodec();
+  private final ClusterTierManagement management;
+
+  private volatile ConcurrentHashMap<String, List<InvalidationTuple>> inflightInvalidations;
```

Comment:
`inflightInvalidations` should now just be `List<InvalidationTuple>` because this entity maps to only one store. This should even simply compute on this variable.